ga4gh/versioning.py

```diff
@@ -0,0 +1,152 @@
+"""
+Classes for implementing and enforcing API versioning
+
+Each route belongs to a RouteGroup.
+Each RouteGroup specifies a set of rules which, when satisfied,
+users are allowed to access its routes.  These rules are:
+    1) a lifecycle
+    2) enabled and disabled versions
+
+Right now, three lifecycle stages are defined:
+    development - the default; endpoints that are under active development;
+        only can be accessed in development mode
+    active - live and stable; able to be accessed in any mode
+    deactivated - not able to be accessed in any mode
+
+Enabled and disabled versions:
+    these two versions specify the min and max versions at which the
+    endpoints can be accessed
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+
+
+class Lifecycle(object):
+    """
+    A lifecycle stage for a route group
+    """
+    # TODO more lifecycle stages?
```

Comment:
1) We could do that.  But there's no way to return any information about the endpoint's lifecycle status to the end user via the protocol, however.  The user would need to read docs somewhere.

2) To differentiate it from ACTIVE, a status which communicates "do not modify these endpoints for any reason because clients now depend on them"

But yes, our lifecycle should be well defined, in any case.